src/condor_dagman/parse.cpp

```diff
@@ -2381,6 +2388,69 @@ parse_node_status_file(
 	return true;
 }
 
+//-----------------------------------------------------------------------------
+// Function: parse_save_point_file
+// Purpose:  Parses a line specifying a node to write a save point file. The
+//           save point file will be written first time we start a node. This
+//           while will be written out as a partial rescue. Users can optionally
+//           designate a filename for the file to be written as. If filename
+//           includes a path then we respect that. Otherwise, we write save file
+//           to save_files subdir created near all other DAG files.
+//-----------------------------------------------------------------------------
+static bool
+parse_save_point_file(Dag *dag, const char* filename, int lineNumber)
+{
+	std::string example = "SAVE_POINT_FILE JobName [Filename]";
+	//Get node name to apply save file info to
+	const char* jobName = strtok(NULL, DELIMITERS);
+	if (jobName == NULL) {
+		debug_printf(DEBUG_QUIET, "ERROR: %s (line %d): No Node specified for SAVE_POINT_FILE.\n",
+					 filename, lineNumber);
+		exampleSyntax(example.c_str());
+		return false;
+	} else if (strcasecmp(jobName,dag->ALL_NODES) == MATCH) { //It is redundant to use all nodes for save files
+		debug_printf(DEBUG_NORMAL, "ERROR: %s (line %d): SAVE_POINT_FILE does not allow ALL_NODES option.\n",
+					 filename, lineNumber);
+		exampleSyntax(example.c_str());
+		return false;
+	}
+
+	const char *jobNameOrig = jobName; // for error output
+	std::string tmpJobName = munge_job_name(jobName);
+	jobName = tmpJobName.c_str();
+
+	//Get save file name/path&name
+	std::string saveFile = strtok(NULL, DELIMITERS) ?: "";
```

Comment:
Note that the elvis operatator is a gcc extension, and not available on VC++.  How about something like
```
const char *p = strtok(nullptr, DELIMITORS);
std::string savefile = p ? p : "";
```